Templates/colors.json

```diff
@@ -0,0 +1,4687 @@
+        { "name": "PixieType[1]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[2]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[3]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[4]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[5]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[6]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[7]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[8]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[9]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[10]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[11]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[12]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[13]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[14]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[15]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[16]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[17]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[18]", "type": "ObjGondolaSpawner::PixieType" },
+        { "name": "PixieType[19]", "type": "ObjGondolaSpawner::PixieType" }
+      ]
+    },
+    "ObjPirateCannonSpawner": {
+      "fields": [
+        { "name": "IsBaseModel", "type": "bool" },
+        { "name": "IsCollision", "type": "bool" },
+        { "name": "Angle", "type": "float32" },
+        { "name": "SearchDistance", "type": "float32" },
+        { "name": "SearchDistanceMin", "type": "float32" },
+        { "name": "SearchDistanceMax", "type": "float32" },
+        { "name": "StartTime", "type": "float32" },
+        { "name": "NumShots", "type": "uint32" },
+        { "name": "ShotInterval", "type": "float32" },
+        { "name": "WaitTime", "type": "float32" },
+        { "name": "StartTime1st", "type": "float32" },
+        { "name": "NumShots1st", "type": "uint32" },
+        { "name": "ShotInterval1st", "type": "float32" },
+        { "name": "WaitTime1st", "type": "float32" },
+        { "name": "StartTime2nd", "type": "float32" },
+        { "name": "NumShots2nd", "type": "uint32" },
+        { "name": "ShotInterval2nd", "type": "float32" },
+        { "name": "WaitTime2nd", "type": "float32" },
+        { "name": "ShotSpeed", "type": "float32" },
+        { "name": "ShotSpecialNo", "type": "uint32" },
+        { "name": "NumRepeat", "type": "uint32" },
+        { "name": "ItemType", "type": "ObjPirateCannonSpawner::ItemType" },
+        { "name": "NumItems", "type": "uint32" },
+        { "name": "ShotOffsetX", "type": "float32" },
+        { "name": "ShotOffsetY", "type": "float32" }
+      ]
+    },
```

Comment:
Parameters named based on in-game functionality and parameters from other objects with similar functionality.
`NumShots` and `ShotSpecialNo` are given custom names that mimic the Sonic Team naming scheme as closely as possible.